internal/nodes/avi_model_advl4_translator.go

```diff
@@ -151,7 +151,25 @@ func (o *AviObjectGraph) ConstructSvcApiL4VsNode(gatewayName, namespace, key str
 				Gateway:              namespace + "/" + gatewayName,
 			},
 		}
-		if lib.GetSEGName() != lib.DEFAULT_GROUP {
+
+		// Set SeGroup configuration either from NsxAlbInfraSetting Spec or
+		// env variable (provided during installation)
+		gwClass, err := lib.GetSvcAPIInformers().GatewayClassInformer.Lister().Get(gw.Spec.GatewayClassName)
+		if err != nil {
+			utils.AviLog.Warnf("Unable to get corresponding GatewayClass %s", err.Error())
+		} else {
+			if gwClass.Spec.ParametersRef != nil && gwClass.Spec.ParametersRef.Group == "ako.vmware.com" && gwClass.Spec.ParametersRef.Kind == "NsxAlbInfraSetting" {
```

Comment:
Can't this check be done by simply using the `controller` info on the GatewayClass?

Reply:
This is handled as part of `isGatewayDelete`